app/celery/tasks.py

```diff
@@ -244,11 +241,12 @@ def save_sms(self, service_id, notification_id, encrypted_notification, sender_i
                 reply_to_text=reply_to_text,
             )
         except IntegrityError:
-            if notification_exists(notification_id):
-                saved_notification = get_notification(notification_id)
-
-            else:
-                raise
+            current_app.logger.warning(
+                f"{NotificationType.SMS}: {notification_id} already exists."
```

Comment:
This will fix the issue where we fail on retrying save_sms, but won't that result in sending the same message multiple times?

Reply:
I see now that this short-circuits things.  It looks like we end up with one retry, and return abruptly instead of doing 5 retries.

Reply:
Thank you both for double checking and confirming this!